### DISCUSSION TITLE
Webserver and webinterface?

### DISCUSSION THREAD

**Discussion by PrototypeSomething** [Zer0-bit/gaggiuino#553] [Q&A]
I've been looking into this project for well over a year now, even before I got my Gaggia and I'm wondering how the webserver is coming along? Does it exist and if it does, how do I pull/push info to it?

My end goal is to use a mobile phone or other microcontroller with a display as a touch screen instead of the wildly expensive and tiny nextion display.

If it could be to any help I would love to help and design a scalable interface for the webserver

**Comment by dcampana** [comment]
I do not know how developed the web app is, but I believe you just use a ESP32 with wifi and select ESP32 in platformIO for compiling and change the pin definitions to align with your new configuration. This is just my knowledge from digging around the code base and also wondering about the web server portion. 

For what it’s worth I threw an ESP32 just flashed with basic firmware in my slightly different espresso machine and closed the lid and I couldn’t access it with BT or WiFi because the case is just grounded metal, your results may vary because you probably have a gaggia that has big vents in the back, but if you really want to go display less I think you would prob need to get a mini-SMI compatible ESP and run an antenna out of the box.

I can’t say I recommend the route of getting a new display, yes the nextion is expensive but making an entirely new UI for a different touch screen and integrating that is probably going to take a LOT of time for you. But hey if you are willing(and having used the nextion I’m down for a better screen) there are TONS of esp32 driven displays that are inexpensive. I think those are probably your best bet at cludging together a quick interface with a much more powerful and cheaper display.

It should be able to connect using the same 2 RX/TX Pins to the STM32, but again you are entering pretty far into the unknown, I do think if you are going to go about this you should probably go with the very ubiquitous esp32 cheap yellow display, there is a huge amount of active community support for them. Best of luck! https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display

Summary because I got wordy: Grab an esp32 hook it up to your computer and flash gaggiuino to it with PlatformIO, the web server is already in the main repo I think you just need to set a build flag.

**Comment by paintflinger** [comment]
> I've been looking into this project for well over a year now, even before I got my Gaggia and I'm wondering how the webserver is coming along? Does it exist and if it does, how do I pull/push info to it?
> 
> My end goal is to use a mobile phone or other microcontroller with a display as a touch screen instead of the wildly expensive and tiny nextion display.
> 
> If it could be to any help I would love to help and design a scalable interface for the webserver

Recommend you join the discord.  You'll see the current status or the web interface and new display.
